keyboards/knops/mini/keymaps/mverteuil/keymap.c

```diff
@@ -0,0 +1,350 @@
+                break;
+            case 2:
+                if (readPin(B7)) {
+                    writePinLow(C6);
+                } else {
+                    writePinLow(C7);
+                }
+                break;
+            case 3:
+                writePinLow(D4);
+                break;
+            case 4:
+                writePinLow(E6);
+                break;
+            case 5:
+                writePinLow(B4);
+                break;
+            case 6:
+                writePinLow(D6);
+                break;
+        }
+    }
+}
+
+void set_layer_led(int layerId, bool layerState) {
+    writePinHigh(D5);
+    writePinLow(B6);
+    writePinHigh(B0);
+
+    if (layerState) {
+        switch (layerId) {
+            case 0:
+                writePinLow(D5);
+                break;
+            case 1:
+                writePinHigh(B6);
+                break;
+            case 2:
+                writePinLow(B0);
+                break;
+            case 3:
+                writePinHigh(D5);
+                writePinLow(B6);
+                writePinHigh(B0);
+                break;
+        }
+    } else {
+        switch (layerId) {
+            case 0:
+                writePinHigh(D5);
+                break;
+            case 1:
+                writePinLow(B6);
+                break;
+            case 2:
+                writePinHigh(B0);
+                break;
+            case 3:
+                writePinLow(D5);
+                writePinHigh(B6);
+                writePinLow(B0);
+                break;
+        }
+    }
+}
+
+/*
+ * This function led_set_layer gets called when you switch between layers.
+ * It allows you to turn on and off leds for each different layer and do
+ * other cool stuff.
+ */
+void led_set_layer(int layer) {
+    switch (layer) {
+        case 0:
+            set_switch_led(1, true);
+            set_switch_led(2, true);
+            set_switch_led(3, true);
+            set_switch_led(4, true);
+            set_switch_led(5, false);
+            set_switch_led(6, false);
+            break;
+
+        case 1:
+            set_switch_led(1, true);
+            set_switch_led(2, true);
+            set_switch_led(3, true);
+            set_switch_led(4, false);
+            set_switch_led(5, true);
+            set_switch_led(6, false);
+            break;
+
+        case 2:
+            set_switch_led(1, true);
+            set_switch_led(2, true);
+            set_switch_led(3, true);
+            set_switch_led(4, false);
+            set_switch_led(5, false);
+            set_switch_led(6, true);
+            break;
+
+        case 3:
+            set_switch_led(1, false);
+            set_switch_led(2, false);
+            set_switch_led(3, true);
+            set_switch_led(4, true);
+            set_switch_led(5, true);
+            set_switch_led(6, true);
+            break;
+    }
+}
+
+void led_init_ports() {
+    writePinHigh(B7);
+
+    DDRB &= ~(1 << 7);
```

Comment:
```suggestion
    setPinInputHigh(B7);
```

Reply:
Thanks! I didn't know what the alias for setting/reading DDRB pins were, it's been a long time since I did any microcontroller stuff and much of the source related to LEDs comes from the default keymap. 

I feel I'd made some minor improvements, but these are way more readable.

Should we also add these improvements to the default keymap, or there code ownership concerns with that?

Now, if I only understood why the initialization code is the way it is... there's no docs from Knops w.r.t. LEDs, so I figured most of it out by trial and error and experimenting/discovery with removing certain calls.

I read something about this device essentially being an Arduino Leonardo, so I wonder if that's where I should be looking...

In any event, I'll integrate these suggestions and thanks again. 👍

Reply:
Ah, I didn't realise this was from the default keymap - that's fine to do the same there, but I think I'd prefer a separate PR for it, as this one's quite large already.